[14.0][OU-ADD] sale: Migration scripts

MiquelRForgeFlow opened the pull request:
Migration done.

MiquelRForgeFlow committed:
[MIG] sale

flotho reviewed (approved):
Successfully tested on 2 production DB migrated from 10.0

pedrobaeza committed:
[OU-FIX] sale: Renamed noupdate XML-IDs should update data

If not, the record rules will have an incorrect domain.

Also avoid to write empty data to non really changed record rules (due
to global field).

TT30024

pedrobaeza reviewed (approved):
Adjusted noupdate data due to the renaming + optimize to not write some non modified record rules.

pedrobaeza commented:
> Successfully tested on 2 production DB migrated from 10.0

If you have such DBs already in production, you would need to adjust manually the record rules for accessing account.move and account.move.line due to the renaming of `type` to `move_type`.
